source/extensions/filters/network/local_ratelimit/local_ratelimit.cc

```diff
@@ -106,15 +122,30 @@ Network::FilterStatus Filter::onNewConnection() {
     config_->stats().rate_limited_.inc();
     ENVOY_CONN_LOG(trace, "local_rate_limit: rate limiting connection",
                    read_callbacks_->connection());
-    read_callbacks_->connection().streamInfo().setResponseFlag(
-        StreamInfo::ResponseFlag::UpstreamRetryLimitExceeded);
-    read_callbacks_->connection().close(Network::ConnectionCloseType::NoFlush);
```

Comment:
The response flag setting should be kept.

Reply:
updated, thanks!